### PR TITLE
Fix access to null pointer in TiffParser

### PR DESCRIPTION
**Pull request by piponazo**
Fix for the second issue reported in #590.

`te->pValue()` could be `0x0` with the toc file provided by the user.

I did not add a test to reproduce the issue because I do not have much spare time lately, and I think the fix is obvious enough.

### CODE REVIEW

File: src/tiffimage_int.cpp [hunk 1]
```diff
@@ -1696,6 +1696,7 @@ namespace Exiv2 {
             pSourceDir->accept(finder);
             TiffEntryBase* te = dynamic_cast<TiffEntryBase*>(finder.result());
             if (   te
+                && te->pValue()
```
Comment (tbeu) [inline src/tiffimage_int.cpp]: I'd prefer to have `te->pValue()` as temporary here.

```cpp
const auto& pV = te != nullptr ? te->pValue() : nullptr;
if (pV && pV->typeID() && ...)
```

Comment (piponazo) [reply src/tiffimage_int.cpp]: I'll do it 😉

Comment (piponazo) [reply src/tiffimage_int.cpp]: done

File: src/tiffimage_int.cpp [hunk 2]
```diff
@@ -1695,10 +1695,8 @@ namespace Exiv2 {
             TiffFinder finder(0x00fe, imageGroups[i]);
             pSourceDir->accept(finder);
             TiffEntryBase* te = dynamic_cast<TiffEntryBase*>(finder.result());
-            if (   te
-                && te->pValue()->typeId() == unsignedLong
-                && te->pValue()->count() == 1
-                && (te->pValue()->toLong() & 1) == 0) {
+            const Value* pV = te != NULL ? te->pValue() : NULL;
```
Comment (tbeu) [inline src/tiffimage_int.cpp]: `nullptr` please.

Comment (piponazo) [reply src/tiffimage_int.cpp]: It cannot be in 0.27, where we are using C++98 😉

Comment (tbeu) [reply src/tiffimage_int.cpp]: Wrong branch I am, sorry.

File: src/tiffimage_int.cpp [hunk 3]
```diff
@@ -1695,10 +1695,8 @@ namespace Exiv2 {
...
+            if (te && pV && pV->typeId() == unsignedLong && pV->count() == 1 && (pV->toLong() & 1) == 0) {
```
Comment (tbeu) [inline src/tiffimage_int.cpp]: No need to check `te` again here.

Comment (piponazo) [reply src/tiffimage_int.cpp]: :man_facepalming:  right

Comment (piponazo) [reply src/tiffimage_int.cpp]: done!

File: src/tiffimage_int.cpp [hunk 4]
```diff
@@ -1695,10 +1695,8 @@ namespace Exiv2 {
             TiffFinder finder(0x00fe, imageGroups[i]);
             pSourceDir->accept(finder);
             TiffEntryBase* te = dynamic_cast<TiffEntryBase*>(finder.result());
-            if (   te
-                && te->pValue()->typeId() == unsignedLong
-                && te->pValue()->count() == 1
-                && (te->pValue()->toLong() & 1) == 0) {
+            const Value* pV = te != NULL ? te->pValue() : NULL;
+            if (te && pV->typeId() == unsignedLong && pV->count() == 1 && (pV->toLong() & 1) == 0) {
```
Comment (tbeu) [inline src/tiffimage_int.cpp]: Please: if (pV && pV->typeId() == unsignedLong && pV->count() == 1 && (pV->toLong() & 1) == 0) {`

**Comment by piponazo** [comment]
@tbeu if you are happy with the changes, could you approve? 😉

**Review by D4N — APPROVED** [review]
Thanks for fixing this!

**Comment by piponazo** [comment]
@D4N Thanks to you for the test! Next time, just push me a bit for doing the testing. Sometimes it's difficult to keep the motivation to do the things right 😉

**Review by tbeu — COMMENTED** [review]
Ok now.

**Pull request merged**